### ISSUE TITLE
Upstream improvements for Tracy fiber capture performance.

### ISSUE THREAD

**Issue by benvanik** [google/iree#9627]
We currently can't set `IREE_TRACING_FEATURE_FIBERS` by default due to performance issues. The Tracy `TRACY_FIBERS` define this guards which enables the fiber support also switches the trace recording implementation from thread-local lock-free queues to a single global locked queue. In basic usage this doesn't matter much and if analyzing program flow it's fine to enable (set by default in `IREE_TRACING_MODE>=3`). When trying to trace multithreaded workloads, though, it's quite a disaster as every thread synchronizes two times per zone with all other threads and single zones can go from 50ns to 200us. This can be tricky to spot as it's just a large slow down applied across the board.

The goal here is to be able to correctly render out fibers in the Tracy UI. Though we can operate without this today it makes it difficult to track concurrent/pipelined workloads; instead of seeing two zones here with active/yield states we'd see all those zones interleaved together on the main thread:
![image](https://user-images.githubusercontent.com/75337/175970800-c50311d7-9645-4e32-bcbf-feef7085c298.png)

As programs get more complex this becomes critical to understanding execution:
![image](https://user-images.githubusercontent.com/75337/175970975-093d6cae-9a09-46a5-aa88-850cba93d511.png)

Here's with `IREE_TRACING_FEATURE_FIBERS` enabled:
![image](https://user-images.githubusercontent.com/75337/175969909-a6b2b927-126c-4ece-9075-2b46f5845e20.png)
And here it is disabled:
![image](https://user-images.githubusercontent.com/75337/175969950-d03c974b-13bf-4d76-8bbd-e6953f534eec.png)
The gaps in the former are the delays introduced by lock/memory contention.

Had a brief chat with wolfpld:
https://discord.com/channels/585214693895962624/585214693895962630/988936251837136968

It looks like this is just a TODO and we could fix it:
```
> some extra machinery for queue selection will be needed for this to work full speed
currently the queues are thread_local, but fibers travel across threads and each needs to have own queue
so the events are serialized, as it was easier to do
basically you would need a level of indirection to select the queue
all can be hooked up to the existing macros
> interesting - so (conceptually) fiber enter could select a queue and stash that into a thread local and then
those could be used for operations on that fiber?
> ah ok I see, the ProducerWrapper s_token is thread_local, and each fiber would need an equivalent
ProducerWrapper but can't rely on thread_local (so probably something on the Profiler would have to track?)
```
